graylog2-web-interface/src/views/components/widgets/CopyToDashboardForm.jsx

```diff
@@ -0,0 +1,85 @@
+// @flow strict
+import React, { useEffect, useState, useCallback } from 'react';
+import { Modal, Button, ListGroup, ListGroupItem } from 'components/graylog';
+import type { DashboardsStoreState } from 'views/stores/DashboardsStore';
+import connect from 'stores/connect';
+import { PaginatedList, SearchForm } from 'components/common';
+import { DashboardsActions, DashboardsStore } from 'views/stores/DashboardsStore';
+
+type Props = {
+  onCancel: () => void,
+  onSubmit: (string, string) => void,
+  widgetId: string,
+  dashboards: DashboardsStoreState,
+};
+
+const CopyToDashboardForm = ({ widgetId, onCancel, dashboards: { list = [], pagination }, onSubmit }: Props) => {
+  const [selectedDashboard, setSelectedDashboard] = useState<?string>(null);
+  const [paginationState, setPaginationState] = useState({ query: '', page: 1, perPage: 5 });
+
+  const handleSearch = useCallback((query) => {
+    setPaginationState({
+      query,
+      ...paginationState,
+    });
+    setSelectedDashboard(null);
+    DashboardsActions.search(query, paginationState.page, paginationState.perPage);
+  }, [paginationState]);
```

Comment:
We should make these consistent with the dependencies of `handlePageChange`, I think.